src/madness/world/worldrmi.cc

```diff
@@ -339,9 +339,9 @@ namespace madness {
         // the worst case is where only one node sends huge messages to every node in the communicator
         // AND it has enough threads to use up all tags
         // NB list::size() is O(1) in c++11, but O(N) in older libstdc++
-        MADNESS_ASSERT(ThreadPool::size() < RMI::RmiTask::unique_tag_period() ||
-                       RMI::task_ptr->hugeq.size() <
-                       std::size_t(RMI::RmiTask::unique_tag_period() / RMI::task_ptr->comm.Get_size()));
+        MADNESS_CHECK(ThreadPool::size() < RMI::RmiTask::unique_tag_period() ||
```

Comment:
ditto: ASSERT or throw exception explicitly (or via macro)

Reply:
I think check would be fine here since it will always stop execution one way or another.  However, I will change to exception.